src/main/java/dk/kb/present/util/DsPresentClient.java

```diff
@@ -1,178 +1,99 @@
+    private final static String CLIENT_URL_EXCEPTION="The client url was not constructed correct";
     private final String serviceURI;
-
-    public static final String PRESENT_SERVER_URL_KEY = ".present.url";
-    public static final String PRESENT_SERVER_HEADERS_KEY = ".present.headers";
-
-    /**
-     * Sub-client for the IIIF endpoints.
-     */
-    public final IiifPresentationApi iiif;
-    /**
-     * Sub-client for the service endpoints.
-     */
-    public final ServiceApi service;
-
-    /**
-     * Headers used for all calls to ds-license.
-     */
-    public Map<String, String> headers;
-
-    /**
-     * Create a client for the remote ds-present service.
-     * <p>
-     * This constructor expects the structure
-     * <pre>
-     * present:
-     *   url: 'http://localhost:9073/ds-present/v1'
-     *   headers:
-     *     # Placeholder for authentication until OAuth2 integration is in place
-     *     # Set this to 'internal_service` on local and devel to bypass license check in ds-present
-     *     - Simulated-OAuth2-Group: anonymous
-     * </pre>
-     * @param config following the outlined structure.
-     * @param headers zero or more maps of headers, which will be set for all calls by this client.
-     *                Note that this compounds with headers specified in {@code config}, with the {@code clientHeaders}
-     *                overriding the headers in the {@code config} in case of collisions.
-     */
-    @SuppressWarnings("JavadocLinkAsPlainText")
-    public DsPresentClient(YAML config, Map<String, String>... headers) {
-        this(config.getString(PRESENT_SERVER_URL_KEY), getAllHeaders(config, headers));
-    }
-
-    /**
-     * Create a client for the remote ds-present service, without special headers.
-     * <p>
-     * Note: Clients created using this constructor has no special header supports. The alternative constructor
-     * {@link DsPresentClient(YAML, Map)} handles both URL and fixed header setup.
-     * @param serviceURI the URI for the service, e.g. {@code https://example.com/ds-present/v1}.
-     * @see DsPresentClient(YAML)
-     */
-    public DsPresentClient(String serviceURI) {
-        this(serviceURI, (Map<String, String>) null);
-    }
-
     
     /**
-     * Create a client for the remote ds-present service.
+     * Creates a client for the remote ds-present service.
      * <p>
-     * When working with YAML configs, it is suggested to use the structure
+     * When working with YAML configs, it is suggested to define the present URI as the structure
      * <pre>
-     * present:
-     *   url: 'http://localhost:9073/ds-present/v1'
-     *   headers:
-     *     # Placeholder for authentication until OAuth2 integration is in place
-     *     # Set this to 'internal_service` on local and devel to bypass license check in ds-present
-     *     - Simulated-OAuth2-Group: anonymous
+     * storage:
```

Comment:
`storage` should be `present` here.

Reply:
fixed. But maybe code should be removed. Constructor no longer takes a yaml, but only the URL to the service.

Reply:
Correct, there is no need to specify anthing about YAML here, good idea to re-write the javadoc all-together